### PR TITLE
added syntax highlighting for emacs

### PR DESCRIPTION
**Pull request by netguy204**
Created an emacs major mode to support capnproto syntax highlighting.

### CODE REVIEW

**Comment by dwrensha** [comment]
Nice! I would definitely use this.

I'll nitpick that some of the highlighting is not completely accurate -- e.g. if `interface` is used as a method parameter as it is [here](https://github.com/sandstorm-io/sandstorm/blob/cc598730b2735dab0438dfbf8e9e2c249b966853/src/sandstorm/hack-session.capnp#L118), then it will still be highlighted as a keyword. But that's a minor point, and getting it right would probably take a lot more work.

**Comment by kentonv** [comment]
Thanks, Brian!

FWIW, my own syntax highlighting for qtcreator has the same problem with the contextual keywords. :)

Since I don't know emacs at all I'll go on David's review here. David, let me know when I should merge this.

Also, Brian, feel free to add yourself to `CONTRIBUTORS`.

File: highlighting/emacs/capnp-mode.el [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+;;;###autoload
+(define-derived-mode capnp-mode prog-mode
+  "capn-mode is a major mode for editing capnp protocol files"
+  :syntax-table capnp--syntax-table
+
+  (setq font-lock-defaults '((capnp--font-lock-keywords)))
+
+
+  (setq mode-name "capnp")
+  (define-key capnp-mode-map [remap comment-dwin] 'capnp-comment-dwim))
```
Comment (dwrensha) [inline highlighting/emacs/capnp-mode.el]: should this be `[remap comment-dwim]`?

**Comment by netguy204** [comment]
@kentonv Thank you! Added.
@dwrensha Thanks for the catch. Fixed.

**Comment by dwrensha** [comment]
Looks good to me.

**Comment by kaos** [comment]
Very nice! thank you for this, Brian :+1:

**Pull request merged**